src/gui/editors/PianoRoll.cpp

```diff
@@ -2114,30 +2109,27 @@ void PianoRoll::mouseMoveEvent( QMouseEvent * me )
 						m_pattern->instrumentTrack()->processInEvent( evt );
 					}
 				}
-				else
+				else if( n->isPlaying() )
 				{
-					if( n->isPlaying() )
-					{
-						// mouse not over this note, stop playing it.
-						m_pattern->instrumentTrack()->pianoModel()->handleKeyRelease( n->key() );
+					// mouse not over this note, stop playing it.
+					m_pattern->instrumentTrack()->pianoModel()->handleKeyRelease( n->key() );
 
-						n->setIsPlaying( false );
-					}
+					n->setIsPlaying( false );
 				}
 
-				// set textfloat visible if we're on a note
-				if( on_note )
-				{
-					s_textFloat->moveGlobal( this,	QPoint( me->x() + 4, me->y() + 16 ) );
-					s_textFloat->show();
-				}
-				else
-				{
-					s_textFloat->hide();
-				}
 
 				--it;
+			}
+
+			// If any note was changed
+			if( firstNote )
+			{
+				// Emit pattern has changed
+				m_pattern->dataChanged();
```

Comment:
This signal is connected to ````quantizeChanged()```` which simply calls ````update()```` to redraw the widget. I would agree this only needs to be triggered once.   
This would explain why the float text no longer flashes with your patch :)